packages/core/echo/echo-schema/src/mutable/manipulation.ts

```diff
@@ -50,3 +52,13 @@ export const updateFieldNameInSchema = (
     ),
   );
 };
+
+export const updateTypenameInSchema = (schema: S.Schema<any>, typename: string): S.Schema<any> => {
+  const existingAnnotation = schema.ast.annotations[ObjectAnnotationId] as ObjectAnnotation;
+  invariant(existingAnnotation, "Schema doesn't have ObjectAnnotation");
```

Comment:
`Missing ${ObjectAnnotationId}`

all errors messages should be as terse (and consistent) as is reasonable. Just enough to give the missing runtime information since the static code context will be apparent).